gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagProcessingEngine.java

```diff
@@ -122,6 +134,7 @@ static class DagProcEngineThread implements Runnable {
     private final DagProcFactory dagProcFactory;
     private final DagManagementStateStore dagManagementStateStore;
     private final DagProcessingEngineMetrics dagProcEngineMetrics;
+    private final List<Class<? extends Exception>> nonRetryableExceptions;
```

Comment:
let's use a predicate as the interface - i.e. a function of `Exception -> boolean`, but you could certainly implement one instance of that Strategy interface in terms of a `List<Class<? extends Exception>>`